src/IO/S3/copyS3File.h

```diff
@@ -43,7 +43,8 @@ void copyS3File(
     BlobStorageLogWriterPtr blob_storage_log,
     const std::optional<std::map<String, String>> & object_metadata = std::nullopt,
     ThreadPoolCallbackRunnerUnsafe<void> schedule_ = {},
-    bool for_disk_s3 = false);
+    bool for_disk_s3 = false,
+    std::shared_ptr<const S3::Client> dest_s3_client = nullptr);
```

Comment:
What does it mean if it's nullptr? Maybe it makes sense to add this argument before `dest_bucket` so all arguments related to the destination would be together?

Reply:
this is what I did the first time, but the API starts looking strange, we are doing native copy but we are sending 2 clients.
E.g. we implement a native copy for the S3 disk, and we will never send a different dest client from the source client. I would need to set `nullptr` in all places for it. So I moved to the back as an optional argument.
I can still change if you think it makes sense, or I can rename to something more explicit like `fallback_dest_s3_client`.

Reply:
In my opinion, there is nothing wrong in passing the same client both as `src_client` and `dest_client`. Just like we can pass the same string both as `src_bucket` and `dest_bucket`.